internal/gengapic/generator.go

```diff
@@ -302,3 +303,36 @@ func (g *generator) nestedName(nested pbinfo.ProtoType) string {
 
 	return name
 }
+
+// autoPopulatedFields returns an array of FieldDescriptorProto pointers for the
+// given MethodDescriptorProto that are specified for auto-population per the
+// following restrictions:
+//
+// * The field is a top-level string field of a unary method's request message.
+// * The field is not annotated with google.api.field_behavior = REQUIRED.
+// * The field name is listed in google.api.publishing.method_settings.auto_populated_fields.
+// * The field is annotated with google.api.field_info.format = UUID4.
+func (g *generator) autoPopulatedFields(servName string, m *descriptor.MethodDescriptorProto) []*descriptor.FieldDescriptorProto {
+	var apfs []string
+	// Find the service config's AutoPopulatedFields entry by method name.
+	for _, s := range g.serviceConfig.GetPublishing().GetMethodSettings() {
+		if s.GetSelector() == g.fqn(m) {
+			apfs = s.AutoPopulatedFields
+			break
+		}
+	}
+	inType := g.descInfo.Type[*m.InputType].(*descriptor.DescriptorProto)
```

Comment:
```suggestion
	inType := g.descInfo.Type[m.GetInputType()].(*descriptor.DescriptorProto)
```
Does this work? we should avoid using direct access/dereferencing b.c they cause `panic`

Reply:
Fixed, thanks!